FEAT : ODYA-207 메인화면 커뮤니티 탭 + 메인 서치 다이얼로그 수정

leeJongMin58 opened the pull request:
### 개요
1. 메인서치  부분 api 연결했습니다
2. 메인 화면 커뮤니티탭 완료 했습니다
  => 아직 데이터가 없어서 테스트는 못했습니다.

### 변경사항

### 관련 지라 및 위키 링크
https://weit.atlassian.net/browse/ODYA-207?atlOrigin=eyJpIjoiYzIyY2MwZWY5ZTA4NDk3ZGJhZjkzMjc2YzMyMjk4ZDAiLCJwIjoiaiJ9
### 리뷰어에게 하고 싶은 말
잘부탁 드립니다

Commit:
feat : PlaceCommunity기능 구현

Commit:
chore : 일단 commit

Commit:
chore : 메인 탭 장소조회 커뮤니티

+ 메인 서치 추가 작업

Commit:
FEAT : 메인 탭 장소조회 커뮤니티

+ 메인 서치 추가 작업

minseonglove commented on data/src/main/java/com/weit/data/repository/auth/LoginRepositoryImpl.kt:
```diff
@@ -33,6 +35,9 @@ class LoginRepositoryImpl @Inject constructor(
     private val userNameDataSource: UserInfoDataSource,
 ) : LoginRepository {
 
+    init {
+    }
+
```
지워주세용

minseonglove commented on presentation/src/main/java/com/weit/presentation/ui/map/search/MainSearchTopSheetFragment.kt:
```diff
@@ -13,34 +14,32 @@ import com.weit.presentation.R
-        viewModel.deleteRecentPlaceSearch(it)
-    }
+    private val recentPlaceSearchAdapter =
+        RecentPlaceSearchAdapter(
+            { deleteWord -> viewModel.deleteRecentPlaceSearch(deleteWord) },
+            { touchWord -> viewModel.onTouchRecentWord(touchWord) }
+        )
```
터치한 대상은 같은데 이름은 RecentPlaceSearch, RecentWord로 나눠져 있군요 하나로 통일하는게 좋겠어요

minseonglove commented on presentation/src/main/java/com/weit/presentation/ui/map/search/MainSearchTopSheetFragment.kt:
```diff
@@ -107,47 +102,67 @@ class MainSearchTopSheetFragment(
         }
     }
 
+    override fun onDestroy() {
+        Log.d("test", "onDestroy")
+        super.onDestroy()
+    }
```
로그 지우기

minseonglove commented on presentation/src/main/java/com/weit/presentation/ui/map/search/MainSearchTopSheetFragment.kt:
```diff
@@ -107,47 +102,67 @@ class MainSearchTopSheetFragment(
             viewModel.changeMainSearchFocus(hasFocus)
         }
+
+        binding.btnPlaceSearchCancel.setOnClickListener {
+            viewModel.setBTNPleaseSearchCancelOnClickListener()
+        }
```
클릭한 결과를 처리하는 메소드기 때문에 setOnClickListener라는 이름은 어울리지 않아요

minseonglove commented on presentation/src/main/java/com/weit/presentation/ui/map/search/MainSearchTopSheetViewModel.kt:
```diff
@@ -25,119 +27,155 @@ class MainSearchTopSheetViewModel @Inject constructor(
 
-    fun setBTNPleaseSearchCancelOnClickListener(hasFocus : Boolean){
+    fun onSearchNewWord(searchTerm: String) {
+        registerPlaceSearchWord(searchTerm)
+        plusRecentPlaceSearchWord(searchTerm)
```
이 두개는 무조건 함께 실행되어야 하고 register다음에 plus가 실행되어야 한다는 제약도 있기 때문에 두개의 메소드로 나누지 않고 하나로 처리하는게 좋아보입니다

minseonglove commented on presentation/src/main/java/com/weit/presentation/ui/searchplace/community/PlaceCommunityViewModel.kt:
```diff
@@ -1,8 +1,93 @@
+            if (result.isSuccess) {
+                val newImages = result.getOrThrow()
+
+                if (newImages.isEmpty()){
+                    return@launch
+                }
+
+                communityLastId = newImages.last().communityId
+                if (newImages.isEmpty()){
+                    onNextImages()
+                }
```
62번 라인을 통과했는데 67번 라인이 실행될수있나요?

minseonglove reviewed (commented):
오 몇개 없어요!

Commit:
FEAT : 리뷰수정 ver1

- 로그 및 쓸모없는 코드 정리
- 변수 이름 통일, 메소드 이름 관리
- "RecentSearchWord" data class로 최근 검색 목록 관리
- 장소상세보기 바텀시트에 터치바 생성

overthename commented on presentation/src/main/java/com/weit/presentation/ui/map/search/MainSearchTopSheetFragment.kt:
```diff
@@ -13,34 +13,32 @@ import com.weit.presentation.R
 
-    private val hotPlaceRankLayoutManager = GridLayoutManager(context, 5, GridLayoutManager.HORIZONTAL, false)
+    private val hotPlaceRankLayoutManager =
+        GridLayoutManager(context, 5, GridLayoutManager.HORIZONTAL, false)
```
xml말고 여기서 layoutmanager를 지정한 이유가 있나요?

leeJongMin58 replied:
이거 xml에서도 매개변수 넣고 정의 할수 있나요?

overthename commented on presentation/src/main/java/com/weit/presentation/ui/map/search/MainSearchTopSheetFragment.kt:
```diff
@@ -107,47 +101,62 @@ class MainSearchTopSheetFragment(
         }
     }
 
+    override fun onDestroy() {
```
이 함수는 왜 넣으셨어요?

leeJongMin58 replied:
삭제하겠습니다...

overthename reviewed (approved):
수고하셨습니다

Commit:
feat : 수정 ver2

leeJongMin58 committed:
Merge branch 'develop' into feature/ODYA-207
